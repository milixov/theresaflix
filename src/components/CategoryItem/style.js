import THEME from '../../style/index';
import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
    container: {
        padding: THEME.DIMEN.SCREEN_PADDING,
        marginHorizontal: 16,
        borderRadius: 8,
        backgroundColor: THEME.COLOR.WHITE,
    },
});
