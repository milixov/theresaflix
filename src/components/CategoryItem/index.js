import React from 'react';
import {Text, TouchableOpacity} from 'react-native';

import {styles} from './style';

const CategoryItem = ({title, onPress}) => {
    return (
        <TouchableOpacity onPress={onPress}>
            <Text style={styles.container}>{title}</Text>
        </TouchableOpacity>
    );
};

export default CategoryItem;
