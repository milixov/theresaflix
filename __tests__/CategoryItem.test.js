import React from 'react';
import CategoryItem from '../src/components/CategoryItem';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<CategoryItem />).toJSON();
    expect(tree).toMatchSnapshot();
});
