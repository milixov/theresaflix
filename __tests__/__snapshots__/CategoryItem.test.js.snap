// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 8,
        "marginHorizontal": 16,
        "padding": 24,
      }
    }
  />
</View>
`;
